bookdown/06-extending-learners.Rmd

```diff
@@ -230,6 +224,22 @@ For `classif.rpart` the following replace `<param_set>` above:
       ps$values = list(xval = 0L)
 ```
 
+This can be written shorter, using the `r ref("paradox::ps")` shortcut:
+```{r 06-extending-learners-003-1, eval = FALSE}
+      ps = ps(
+        minsplit = p_int(lower = 1L, default = 20L, tags = "train"),
+        minbucket = p_int(lower = 1L, tags = "train"),
+        cp = p_dbl(lower = 0, upper = 1, default = 0.01, tags = "train"),
+        maxcompete = p_int(lower = 0L, default = 4L, tags = "train"),
+        maxsurrogate = p_int(lower = 0L, default = 5L, tags = "train"),
+        maxdepth = p_int(lower = 1L, upper = 30L, default = 30L, tags = "train"),
+        usesurrogate = p_int(lower = 0L, upper = 2L, default = 2L, tags = "train"),
+        surrogatestyle = p_int(lower = 0L, upper = 1L, default = 0L, tags = "train"),
+        xval = p_int(lower = 0L, default = 0L, tags = "train"),
+        keep_model = p_lgl(default = FALSE, tags = "train")
+      )
+```
+
```

Comment:
Hmm, even if i like the shortcut way I don't know if we want to motivate people to use both ways when defining a learner. I am afraid of inconsistencies.

Reply:
I believe it is useful for large parameter sets also when defining a Learner (or other object) ParamSet; most people will probably not write a learner to be committed to mlr3 for ever, but they will write something quickly to try something. And even for legitimate packages I think it is useful and people should be aware of the options: [This](https://github.com/mb706/mlrintermbo/blob/e0779569a110c6b4c559b77adfbcf9640fdc0e06/R/paramset.R) for example would have been about half as long using the shortcut notation.

Reply:
What are concrete objections to `ps()` here? I can add them to the paragraph if there are pros and cons.

Reply:
Outside mlr3verse everyone is free to use whatever they want. My objections are less against `ps()` but more against doing the same thing in two different ways in terms of code style and maintainability. If there will ever be a change in the API we run into the risk in having to change two versions instead of just one. So we should just decide how to deal with it internally. As the book will be also read by "internal" people it might be worth to mention the code style - once we decided that.

Reply:
I second that we stick to old code style for now. If somebody changes all learners to the new style, we can add this paragraph again.